src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombinationUtil.java

```diff
@@ -77,4 +88,58 @@ public void validateParams(final Map<String, Object> actualParams, final Set<Str
     public float getWeightForSubQuery(final List<Float> weights, final int indexOfSubQuery) {
         return indexOfSubQuery < weights.size() ? weights.get(indexOfSubQuery) : 1.0f;
     }
+
+    /**
+     * Check if number of weights matches number of queries. This does not apply for case when
+     * weights were not provided, as this is valid default value
+     * @param scores collection of scores from all sub-queries of a single hybrid search query
+     * @param weights score combination weights that are defined as part of search result processor
+     */
+    protected void validateIfWeightsMatchScores(final float[] scores, final List<Float> weights) {
+        if (weights.isEmpty()) {
+            return;
+        }
+        if (scores.length != weights.size()) {
+            log.error(
+                String.format(
+                    Locale.ROOT,
+                    "number of weights [%d] must match number of sub-queries [%d] in hybrid query",
+                    weights.size(),
+                    scores.length
+                )
+            );
+            throw new IllegalArgumentException("number of weights must match number of sub-queries in hybrid query");
```

Comment:
Returning size of weights and queries should be fine here. So security concern.

Reply:
Yup, I do it as a rule of thumb, but I guess that's ok here. Let me push a commit